File: erp-app/src/app/modules/accounting/models/treasury-search.dto.ts
import { AccountTypeEnum } from "src/app/shared/enums/account-type.enum";
import { PaymentMethodEnum } from "src/app/shared/enums/payment-method.enum";
import { PagingDTO } from "src/app/shared/models/paging-dto";

export class TreasurySearchDTO extends PagingDTO {
    date: string;
    dateFrom: string | undefined;
    dateTo: string | undefined;
    accountTypeId: AccountTypeEnum | null;
    clientVendorId: number | null;
    transactionTypeId: AccountTypeEnum | null;
    paymentMethodId: PaymentMethodEnum | null;
    amount: number | null;
    refNo: string;
    bankAccountNo: string;
    checkNo: string;
    beneficiaryName: string;
}

File: ERPServer/Shared/Shared/Entities/Accouting/TreasurySearchDTO.cs
using Shared.Entities.Shared;
using Shared.Enums;
using System;

namespace Shared.Entities.Accouting
{
    public class TreasurySearchDTO : Paging
    {
        public string Date { get; set; }
        public AccountTypeEnum? AccountTypeId { get; set; }
        public long? ClientVendorId { get; set; }
        public TransactionTypeEnum? TransactionTypeId { get; set; }
        public PaymentMethodEnum? PaymentMethodId { get; set; }
        public string RefNo { get; set; }
        public string BankAccountNo { get; set; }
        public string CheckNo { get; set; }
        public string Notes { get; set; }
    }
}
